### PR TITLE
Static analysis fixes

### PR DESCRIPTION
**Pull request by PastaPastaPasta**
See https://github.com/PastaPastaPasta/dash/commit/f3733fe807c82cfbcaf285a231d51d64f452a8af for the script with static analysis / linting that I used.

Note that running it requires https://github.com/PastaPastaPasta/dash/commit/baf6e1b7aab70a5d208fde44f8a782959138da10 for reasons

### CODE REVIEW

File: src/llmq/quorums.cpp [hunk 1]
```diff
@@ -381,11 +381,11 @@ bool CQuorumManager::HasQuorum(Consensus::LLMQType llmqType, const uint256& quor
 
 bool CQuorumManager::RequestQuorumData(CNode* pFrom, Consensus::LLMQType llmqType, const CBlockIndex* pQuorumIndex, uint16_t nDataMask, const uint256& proTxHash) const
 {
-    if (pFrom->nVersion < LLMQ_DATA_MESSAGES_VERSION) {
+    if (pFrom == nullptr || pFrom->nVersion < LLMQ_DATA_MESSAGES_VERSION) {
```
Comment (UdjinM6) [inline src/llmq/quorums.cpp:384]: pls split `pFrom == nullptr` case into its own `if`

File: src/llmq/quorums_chainlocks.cpp [hunk 2]
```diff
@@ -35,17 +35,14 @@ std::string CChainLockSig::ToString() const
 
 CChainLocksHandler::CChainLocksHandler()
 {
-    scheduler = new CScheduler();
-    CScheduler::Function serviceLoop = boost::bind(&CScheduler::serviceQueue, scheduler);
-    scheduler_thread = new boost::thread(boost::bind(&TraceThread<CScheduler::Function>, "cl-schdlr", serviceLoop));
+    scheduler = std::make_unique<CScheduler>();
+    CScheduler::Function serviceLoop = std::bind(&CScheduler::serviceQueue, scheduler.get());
+    scheduler_thread = std::make_unique<std::thread>(std::bind(&TraceThread<CScheduler::Function>, "cl-schdlr", serviceLoop));
```
Comment (UdjinM6) [inline src/llmq/quorums_chainlocks.cpp:40]: this violates usage notes in `scheduler.h`, switching from boost to std for schedulers should be done in the corresponding backport imo

File: src/rpc/rpcevo.cpp [hunk 3]
```diff
@@ -942,7 +941,6 @@ static UniValue BuildDMNListEntry(CWallet* pwallet, const CDeterministicMNCPtr&
     if (pwallet) {
         UniValue walletObj(UniValue::VOBJ);
         walletObj.pushKV("hasOwnerKey", hasOwnerKey);
-        walletObj.pushKV("hasOperatorKey", hasOperatorKey);
```
Comment (UdjinM6) [inline src/rpc/rpcevo.cpp:945]: this was kept for backwards (and maybe future) compatibility iirc, I would suggest to still keep it but simply as `walletObj.pushKV("hasOperatorKey", false);` for now maybe

**Review by UdjinM6 — CHANGES_REQUESTED** [review]
Nice! 👍 

pls see few notes below

File: src/rpc/rpcevo.cpp (hunk 3 above)
Comment (PastaPastaPasta) [reply src/rpc/rpcevo.cpp:945]: I'm not really sure this is needed, but sure, done

File: src/llmq/quorums_chainlocks.cpp (hunk 2 above)
Comment (PastaPastaPasta) [reply src/llmq/quorums_chainlocks.cpp]: reverted

File: src/llmq/quorums.cpp (hunk 1 above)
Comment (PastaPastaPasta) [reply src/llmq/quorums.cpp]: done

**Review by UdjinM6 — APPROVED** [review]
LGTM, utACK

**Pull request merged**